cirrus/server/cirrus/main.py

```diff
@@ -27,5 +44,39 @@ async def compute_enrollment_status():
     client_feature_configuration = fml.compute_feature_configurations(
         enrolled_partial_configuration, feature_configurations
     )
-
     return client_feature_configuration
+
+
+@app.on_event("startup")
+async def start_scheduler():
+    scheduler.start()
+    app.state.scheduler = scheduler
+
+
+@app.on_event("shutdown")
+async def shutdown_scheduler():
+    scheduler.shutdown()
+
+
+async def fetch_schedule_recipes() -> None:
+    try:
+        remote_setting.fetch_recipes()
+    except Exception as e:
+        # If an exception is raised, log the error and schedule a retry
+        logger.error(f"Failed to fetch recipes: {e}")
+        scheduler.add_job(  # type: ignore
+            fetch_schedule_recipes,
+            "interval",
+            seconds=remote_setting_refresh_rate_in_seconds,
+            max_instances=1,
+            max_retries=3,
+        )
```

Comment:
Why do you need to schedule a retry like this here?  Especially using the globally defined refresh rate?  Isn't this redundant with the interval scheduler below?

Reply:
so this max retries/ scheduler job is optional, the idea is let's say global rate is 1 hour and so if the job fails for some reason it will only try after 1 hour (the outside scheduler will do that), but having this job schedule (inside a function when exception is raised) will help us to retry, technically the refresh rate of this should be different than the global refresh rate and will only try failed job 3 times